core/test/AWS/Lambda/Event/APIGatewayV2Spec.hs

```diff
@@ -41,62 +42,103 @@ testDecodeTextBodyRequestFromFile :: Expectation
   reqBytes <- LB.readFile fname
-  let Just req = decode @(HTTPRequest MockBody) reqBytes
-  req `shouldBe` mockRequest (MockBody "bodyval" 123)
+  let Just req = decode @HTTPRequest reqBytes
+  req `shouldBe` mockRequest "{\"bodykey1\":\"bodyval\",\"bodykey2\":123}"
+  requestBodyJSON @MockBody req `shouldBe` Right (MockBody "bodyval" 123)
 
 
 testDecodeBase64BodyRequestFromFile :: Expectation
 testDecodeBase64BodyRequestFromFile = do
   fname <- Paths.getDataFileName "api-gateway-v2-request-base64-body.json"
   reqBytes <- LB.readFile fname
-  let Just req = decode @(HTTPRequest Base64EncodedBody) reqBytes
-  req `shouldBe` (mockRequest (Base64EncodedBody "base64val") & isBase64Encoded .~ True)
+  let Just req = decode @HTTPRequest reqBytes
+  req `shouldBe` (mockRequest "YmFzZTY0dmFs" & isBase64Encoded .~ True)
+  requestBodyText req `shouldBe` Right "base64val"
+
+
+testDecodeMinimalRequestFromFile :: Expectation
+testDecodeMinimalRequestFromFile = do
+  fname <- Paths.getDataFileName "api-gateway-v2-request-minimal.json"
+  reqBytes <- LB.readFile fname
+  let Just req = decode @HTTPRequest reqBytes
+  req `shouldBe` mockMinimalRequest
 
 
 testDecodeTextBodyResponseFromFile :: Expectation
 testDecodeTextBodyResponseFromFile = do
   fname <- Paths.getDataFileName "api-gateway-v2-response-text-body.json"
   reqBytes <- LB.readFile fname
-  let Just req = decode @(HTTPResponse Text) reqBytes
+  let Just req = decode @HTTPResponse reqBytes
   req `shouldBe` mockResponse "Hello from Lambda!"
 
 
 testDecodeJsonBodyResponseFromFile :: Expectation
 testDecodeJsonBodyResponseFromFile = do
   fname <- Paths.getDataFileName "api-gateway-v2-response-json-body.json"
   reqBytes <- LB.readFile fname
-  let Just req = decode @(HTTPResponse MockBody) reqBytes
-  req `shouldBe` mockResponse (MockBody "bodyval" 123)
+  let Just req = decode @HTTPResponse reqBytes
+  req `shouldBe` mockResponse "{\"bodykey1\":\"bodyval\",\"bodykey2\":123}"
 
 
 testDecodeBase64BodyResponseFromFile :: Expectation
 testDecodeBase64BodyResponseFromFile = do
   fname <- Paths.getDataFileName "api-gateway-v2-response-base64-body.json"
   reqBytes <- LB.readFile fname
-  let Just req = decode @(HTTPResponse Base64EncodedBody) reqBytes
-  req `shouldBe` (mockResponse (Base64EncodedBody "base64val") & isBase64Encoded ?~ True)
+  let Just req = decode @HTTPResponse reqBytes
+  req `shouldBe` (mockResponse "YmFzZTY0dmFs" & isBase64Encoded ?~ True)
+
+
+mockMinimalRequest :: HTTPRequest
+mockMinimalRequest = HTTPRequest
+  { _version = "2.0"
+  , _routeKey = "ANY /"
+  , _rawPath = "/"
+  , _rawQueryString = ""
+  , _cookies = Nothing
+  , _headers = Map.fromList [("accept","*/*"),("content-length","0"),("host","00aabbccdd.execute-api.us-east-1.amazonaws.com"),("user-agent","curl/7.70.0"),("x-amzn-trace-id","Root=1-5ebd8107-98d5198d2e2923afe514a5e3"),("x-forwarded-for","11.11.11.11"),("x-forwarded-port","443"),("x-forwarded-proto","https")]
```

Comment:
Do the incoming headers include the ones that are repeated elsewhere on the request (e.g., User-Agent)?

Reply:
This was actually captured from a real API GW request, just obfuscated sensitive fields.
The message format is weird and repetitive but yeath, headers seems to include them. You can see full request in the test file it loads.

Reply:
It bothers me when there are multiple potential sources of truth, but we wrap the API we have, not the API we want.